pkg/kubelet/kuberuntime/util/util_test.go

```diff
@@ -221,11 +239,84 @@ func TestNamespacesForPod(t *testing.T) {
 				Pid:     runtimeapi.NamespaceMode_CONTAINER,
 			},
 		},
+		"hostUsers: false and feature enabled": {
+			input: &v1.Pod{
+				Spec: v1.PodSpec{
+					HostUsers: &[]bool{false}[0],
+				},
+			},
+			usernsEnabled: true,
+			runtimeHandlers: map[string]kubecontainer.RuntimeHandler{
+				"": {
+					SupportsUserNamespaces: true,
+				},
+			},
+			expected: &runtimeapi.NamespaceOption{
+				Ipc:     runtimeapi.NamespaceMode_POD,
+				Network: runtimeapi.NamespaceMode_POD,
+				Pid:     runtimeapi.NamespaceMode_CONTAINER,
+				UsernsOptions: &runtimeapi.UserNamespace{
+					Mode: runtimeapi.NamespaceMode_POD,
+					Uids: []*runtimeapi.IDMapping{usernsIDs},
+					Gids: []*runtimeapi.IDMapping{usernsIDs},
+				},
+			},
+		},
+		// The hostUsers field can't be set to any value if the feature is disabled.
+		"hostUsers: false and feature disabled --> error": {
+			input: &v1.Pod{
+				Spec: v1.PodSpec{
+					HostUsers: &[]bool{false}[0],
+				},
+			},
+			usernsEnabled: false,
+			expErr:        true,
+		},
+		// The hostUsers field can't be set to any value if the feature is disabled.
+		"hostUsers: true and feature disabled --> error": {
+			input: &v1.Pod{
+				Spec: v1.PodSpec{
+					HostUsers: &[]bool{true}[0],
+				},
+			},
+			usernsEnabled: false,
+			expErr:        true,
+		},
+		"error if runtime handler not found": {
+			input: &v1.Pod{
+				Spec: v1.PodSpec{
+					HostUsers: &[]bool{false}[0],
+				},
+			},
+			usernsEnabled: true,
+			runtimeHandlers: map[string]kubecontainer.RuntimeHandler{
+				"other": {},
+			},
+			expErr: true,
+		},
+		"error if runtime handler does not support userns": {
+			input: &v1.Pod{
+				Spec: v1.PodSpec{
+					HostUsers: &[]bool{false}[0],
+				},
+			},
+			usernsEnabled: true,
+			expErr:        true,
+		},
 	} {
 		t.Run(desc, func(t *testing.T) {
-			actual, err := NamespacesForPod(test.input, &kubecontainertest.FakeRuntimeHelper{}, nil)
-			require.NoError(t, err)
-			require.Equal(t, test.expected, actual)
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, pkgfeatures.UserNamespacesSupport, test.usernsEnabled)()
+
+			fakeRuntimeHelper := kubecontainertest.FakeRuntimeHelper{
+				RuntimeHandlers: test.runtimeHandlers,
+			}
+			actual, err := NamespacesForPod(test.input, &fakeRuntimeHelper, &fakeRuntimeHandlerResolver{})
+			if test.expErr {
```

Comment:
I'd probably add a bit more here. Can we verify the kind of errors we are getting?

Reply:
@kannon92 what do you mean with the kind of error?

What we get here is from a mock, so I don't think it is super useful to test that the mock error contains some specific string or something. That doesn't add any real value, IMHO.

Reply:
I see. I just meant that we verify what kind of error we get but I didn't realize that the errors are mocked errors.

/lgtm

Reply:
Thanks for the review!